packages/next/client/future/image.tsx

```diff
@@ -358,7 +358,14 @@ const ImageElement = ({
         loading={loading}
         style={{ ...imgStyle, ...blurStyle }}
         ref={useCallback(
-          (img: ImgElementWithDataProp) => {
+          (img: ImgElementWithDataProp | null) => {
+            if (img && onError) {
```

Comment:
Why is this conditioned on `onError` only? can't a custom onLoad be provided too?

Reply:
I was specifically concerned about onError because we already have a workaround for onLoad which is onLoadingComplete.

Reply:
Ah, i glanced at it, seems like it is doing quite a bit of stuff. If it were simply dealing with the load stuff it would be nice to unify with the error path here. Is all that complexity intrinsically necessary or is it b/c working around onLoad for hydration was a pain?

Reply:
`onLoadingComplete` does a few things:

- ensures the image is fully decoded 
- removes blur placeholder (not always necessary when svg background used but still necessary for css filter)
- guards against the react hydration issue since it runs even when `onLoad` was never called

Reply:
does it run `onLoad` though? if the user provides a custom onLoad and onError to every img but only the onErrors get called for errors the symmetry is still off?

Reply:
No the user must provide `onLoadingComplete`.

We have always discouraged `onLoad` usage in both `next/image` and `next/future/image` for the 3 reasons above.

If we can solve all the corner cases for svg placeholder, we can probably start supporting `onLoad` in the future.